File: Webfuel.Api/Api/User/UserLoginApi.cs
using MediatR;
using Microsoft.AspNetCore.Mvc;
using Webfuel.Domain;

namespace Webfuel.App
{
    [ApiService]
    [ApiDataSource]
    public static class UserLoginApi
    {
        public static void RegisterEndpoints(IEndpointRouteBuilder app)
        {
            // Commands

            app.MapPost("api/user/login", Login);

            app.MapPost("api/user/change-password", ChangePassword)
                .RequireIdentity();

            app.MapPost("api/user/update-password", UpdatePassword)
                .RequireIdentity();

            app.MapPost("api/user/send-password-reset-email", SendPasswordResetEmail);

            app.MapPost("api/user/reset-password", ResetPassword);

            // Querys

            app.MapPost("api/user/login/query", Query)
                .RequireIdentity();
        }

        public static Task<StringResult> Login([FromBody] LoginUser command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task ChangePassword([FromBody] ChangeUserPassword command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task UpdatePassword([FromBody] UpdateUserPassword command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task SendPasswordResetEmail([FromBody] SendUserPasswordResetEmail command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task ResetPassword([FromBody] ResetUserPassword command, IMediator mediator)
        {
            return mediator.Send(command);
        }

        public static Task<QueryResult<UserLogin>> Query([FromBody] QueryUserLogin command, IMediator mediator)
        {
            return mediator.Send(command);
        }
    }
}

File: Webfuel.App/ClientApp/src/app/api/user-login.api.ts
import { EventEmitter, Injectable, inject } from '@angular/core';
import { Observable, tap } from 'rxjs';
import { ApiService, ApiOptions } from '../core/api.service';
import { ActivatedRouteSnapshot, ResolveFn, RouterStateSnapshot } from '@angular/router';
import { IDataSource } from 'shared/common/data-source';
import { LoginUser, StringResult, ChangeUserPassword, UpdateUserPassword, SendUserPasswordResetEmail, ResetUserPassword, QueryUserLogin, QueryResult, UserLogin } from './api.types';

@Injectable()
export class UserLoginApi implements IDataSource<UserLogin, QueryUserLogin, any, any> {
    constructor(private apiService: ApiService) { }
    
    public login (body: LoginUser, options?: ApiOptions): Observable<StringResult> {
        return this.apiService.request<LoginUser, StringResult>("POST", "api/user/login", body, options);
    }
    
    public changePassword (body: ChangeUserPassword, options?: ApiOptions): Observable<any> {
        return this.apiService.request<ChangeUserPassword, any>("POST", "api/user/change-password", body, options);
    }
    
    public updatePassword (body: UpdateUserPassword, options?: ApiOptions): Observable<any> {
        return this.apiService.request<UpdateUserPassword, any>("POST", "api/user/update-password", body, options);
    }
    
    public sendPasswordResetEmail (body: SendUserPasswordResetEmail, options?: ApiOptions): Observable<any> {
        return this.apiService.request<SendUserPasswordResetEmail, any>("POST", "api/user/send-password-reset-email", body, options);
    }
    
    public resetPassword (body: ResetUserPassword, options?: ApiOptions): Observable<any> {
        return this.apiService.request<ResetUserPassword, any>("POST", "api/user/reset-password", body, options);
    }
    
    public query (body: QueryUserLogin, options?: ApiOptions): Observable<QueryResult<UserLogin>> {
        return this.apiService.request<QueryUserLogin, QueryResult<UserLogin>>("POST", "api/user/login/query", body, options);
    }
    
    changed = new EventEmitter<any>();
}

